src/components/ui4/RowUi4.tsx

```diff
@@ -0,0 +1,155 @@
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const { body, title, children, maximumHeight = 'medium', error, icon, onLongPress, onPress } = props
+  const { type = onLongPress == null && onPress == null ? 'default' : 'touchable' } = props
+
+  const numberOfLines = textHeights[maximumHeight]
+
+  const handlePress = useHandler(async () => {
+    if (type === 'copy' && body != null) {
+      triggerHaptic('impactLight')
+      Clipboard.setString(body)
+      showToast(lstrings.fragment_copied)
+    } else if (onPress != null) {
+      triggerHaptic('impactLight')
+      try {
+        await onPress()
+      } catch (err) {
+        showError(err)
+      }
+    }
+  })
+
+  const handleLongPress = useHandler(async () => {
+    if (onLongPress != null) {
+      triggerHaptic('impactLight')
+      try {
+        await onLongPress()
+      } catch (err) {
+        showError(err)
+      }
+    }
+  })
+
+  return type === 'loading' ? (
+    <View style={styles.container}>
+      <View style={styles.content}>
+        <EdgeText style={styles.textHeader}>{title}</EdgeText>
+        <ActivityIndicator style={styles.loader} color={theme.primaryText} size="large" />
+      </View>
+    </View>
+  ) : (
+    <TouchableHighlight
+      accessible={false}
+      onPress={handlePress}
+      onLongPress={handleLongPress}
+      disabled={handlePress == null && handleLongPress == null}
+      underlayColor={theme.touchHighlightUi4}
+    >
+      <View style={styles.container}>
+        {icon && <View style={styles.iconContainer}>{icon}</View>}
+        <View style={styles.content}>
+          {title == null ? null : (
+            <EdgeText disableFontScaling ellipsizeMode="tail" style={error ? styles.textHeaderError : styles.textHeader}>
+              {title}
+            </EdgeText>
+          )}
+          {children == null
+            ? typeof body === 'string' && (
+                <EdgeText style={styles.textBody} numberOfLines={numberOfLines} ellipsizeMode="tail">
+                  {body}
+                </EdgeText>
+              )
+            : children}
+        </View>
+        <View style={styles.rightButtonContainer}>
+          {type === 'touchable' && <FontAwesome5 name="chevron-right" style={styles.tappableIcon} size={theme.rem(1)} />}
+          {type === 'editable' && <FontAwesomeIcon name="edit" style={styles.tappableIcon} size={theme.rem(1)} />}
+          {type === 'copy' && <FontAwesomeIcon name="copy" style={styles.tappableIcon} size={theme.rem(1)} />}
+          {type === 'delete' && <FontAwesomeIcon name="times" style={styles.tappableIcon} size={theme.rem(1)} />}
+          {type === 'questionable' && <SimpleLineIcons name="question" style={styles.tappableIcon} size={theme.rem(1)} />}
+        </View>
+      </View>
+    </TouchableHighlight>
+  )
+}
+
+// TODO: Adjust margin/padding so everything combines with correct layout no
+// matter the combination of UI4 components.
```

Comment:
Ambitious. I assume this will happen later as more stuff gets built out?

Reply:
Yes, I tried to start doing it and then found I had to make more changes upon tackling subsequent tasks. Figured it would be better to just handle it at the end to make sure everything works and avoid all the rebasing, it's mostly fine right now.